File: CJComplexUIKitDemo/CJComplexUIKitDemo/Modules/LinkedMenuDemo/DataSource/GuideMenuDataSource.m
//
//  GuideMenuDataSource.m
//  CJComplexUIKitDemo
//
//  Created by ciyouzen on 2017/2/25.
//  Copyright © 2017年 dvlproad. All rights reserved.
//

#import "GuideMenuDataSource.h"
#import "GuideMenuTableViewCell.h"
#import "CQTSLocImagesUtil.h"

@interface GuideMenuDataSource () {
    
}
@property (nonatomic, strong) NSArray<CQTSLocImageDataModel *> *dataModels;    /**< 菜单数据 */

@property (nonatomic, copy, readonly) void(^cellAtIndexPathConfigBlock)(UICollectionViewCell *bCollectionViewCell, NSIndexPath *bIndexPath); /**< 绘制指定indexPath的cell */

@end


@implementation GuideMenuDataSource

#pragma mark - Init
/*
 *  初始化 TableView 的 dataSource
 *
 *  @param rowCount             菜单个数
 *
 *  @return TableView 的 dataSource
 */
- (instancetype)initWithRowCount:(NSInteger)rowCount
{
    NSInteger iRowCount = rowCount;
    
    NSMutableArray<CQTSLocImageDataModel *> *dataModels = [CQTSLocImagesUtil __getTestLocalImageDataModelsWithCount:iRowCount randomOrder:NO];
    for (int row = 0; row < iRowCount; row++) {
        CQTSLocImageDataModel *module = [dataModels objectAtIndex:row];
        module.name = [NSString stringWithFormat:@"%02d", row];
    }

    self = [self initWithDataModels:dataModels];
    if (self) {
        
    }
    return self;
}

/*
 *  初始化 TableView 的 dataSource
 *
 *  @param dataModels            菜单列表数据
 *
 *  @return TableView 的 dataSource
 */
- (instancetype)initWithDataModels:(NSArray<CQTSLocImageDataModel *> *)dataModels
{
    self = [super init];
    if (self) {
        self.dataModels = dataModels;
    }
    return self;
}


/*
 *  注册 TableView 所需的所有 cell
 */
- (void)registerAllCellsForTableView:(UITableView *)tableView {
    [tableView registerClass:[GuideMenuTableViewCell class] forCellReuseIdentifier:@"cell"];
}

#pragma mark - UITableViewDataSource

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataModels.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    CQTSLocImageDataModel *moduleModel = [self.dataModels objectAtIndex:indexPath.row];
    
    GuideMenuTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    cell.myImageView.image = moduleModel.image;
    cell.mainTitleLabel.text = moduleModel.name;
    
    return cell;
}

- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
    return nil;
}

@end

File: CJListDemo-Swift/LinkedMenu/GuideMenuDataSource.swift
//
//  GuideMenuDataSource.swift
//  CJListDemo-Swift
//
//  Created by qian on 2025/2/10.
//

import UIKit
import CQDemoKit

@objc public class GuideMenuDataSource: NSObject, UITableViewDataSource {
    public var dataModels: [CQTSLocImageDataModel] = [] // 菜单数据
    
    /*
    // 初始化 TableView 的 dataSource
    init(rowCount: Int) {
        let dataModels: [CQTSLocImageDataModel] = CQTSLocImagesUtil.__getTestLocalImageDataModels(withCount: rowCount, randomOrder: false) as! [CQTSLocImageDataModel]
        
        for (index, module) in dataModels.enumerated() {
            module.name = String(format: "%02d", index)
        }
        
        self.dataModels = dataModels
    }
    
    // 初始化 TableView 的 dataSource
    init(dataModels: [CQTSLocImageDataModel]) {
        self.dataModels = dataModels
    }
    */
    
    // 注册 TableView 所需的所有 cell
    func registerAllCells(for tableView: UITableView) {
        tableView.register(GuideMenuTableViewCell.self, forCellReuseIdentifier: "cell")
    }
    
    // MARK: - UITableViewDataSource
    
    public func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataModels.count
    }
    
    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let moduleModel = dataModels[indexPath.row]
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as! GuideMenuTableViewCell
        let image = UIImage.cqdemokit_xcassetImageNamed(moduleModel.imageName)
        cell.myImageView.image = image
        cell.mainTitleLabel.text = moduleModel.name
        
        return cell
    }
    
    public func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
        return nil
    }
}

